backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/projection/impl/LuceneProjectionTypeKeys.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.search.projection.impl;
+
+import static org.hibernate.search.engine.search.projection.spi.ProjectionTypeKeys.key;
+
+import org.hibernate.search.engine.search.common.spi.SearchQueryElementTypeKey;
+import org.hibernate.search.engine.search.projection.spi.CompositeProjectionBuilder;
+
+public final class LuceneProjectionTypeKeys {
+
+	private LuceneProjectionTypeKeys() {
+	}
+
+	public static final SearchQueryElementTypeKey<CompositeProjectionBuilder> DOCUMENT = key( "document" );
```

Comment:
Since you don't use the builder, you might as well hide its type (`CompositeProjectionBuilder` is incorrect anyway):

```suggestion
	public static final SearchQueryElementTypeKey<?> DOCUMENT = key( "document" );
```